.github/workflows/deploy.yml

```diff
@@ -0,0 +1,67 @@
+name: Deploy preview site
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: [$default-branch]
```

Comment:
This maybe not a legal value, it looks like this workflow is not triggered automatically

Reply:
Whoops, it only works in Workflow templates... I copied it from a template, it should be changed to `master` here.

Ref https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable